### PR TITLE
fix(remix-server-runtime): fix headers function access to loaderHeaders when parent has no loader

### PR DESCRIPTION
**Pull request by xstevenyung**
This should fixes issue: https://github.com/remix-run/remix/issues/1140

The issue came from `routeLoaderResponses` not taking into account that some parent route might not have loaders hence no response. Because `getDocumentHeaders` matched the response to the `index` position in `routeLoaderResponses`, this could provoke a mismatch.

I replaced `routeLoaderResponses` from an array to a `Record<string, Response>` (matching the route's id instead of the index) which would solve the issue.

I also created a test but I think it might not be in the right place, feel free to move it if needed.

Thanks for your awesome work, I love Remix !

### CODE REVIEW

File: packages/remix-server-runtime/__tests__/server-test.ts [hunk 1]
```diff
@@ -1679,4 +1679,37 @@ describe("shared server runtime", () => {
       expect(spy.console.mock.calls.length).toBe(1);
     });
   });
+
+  describe("header requests", () => {
+    test.only("can access loader headers if parent doesn't have a loader itself", async () => {
```
Comment (mkrtchian) [inline packages/remix-server-runtime/__tests__/server-test.ts:1684]: I think you forgot to remove this `only`.

Comment (xstevenyung) [reply packages/remix-server-runtime/__tests__/server-test.ts]: Nice catch, fixed. Thanks very much for commenting 👍

**Comment by ryanflorence** [comment]
Awesome! you mind removing your test and then simply removing the `skip` on this one? (and double check my work from last week that this is testing the same thing?)

https://github.com/remix-run/remix/blob/40216cdf83593b53e1f8d98b936e4662238787b6/integration/headers-test.ts#L83

**Comment by ryanflorence** [comment]
nvm I got it :)

**Pull request merged**